IdentityCore/src/oauth2/aad_base/MSIDAADWebviewFactory.m

```diff
@@ -111,13 +111,24 @@ @implementation MSIDAADWebviewFactory
         platformParams = [[MSIDWebViewPlatformParams alloc] initWithExternalSSOContext:configuration.ssoContext];
     }
     
-     MSIDAADOAuthEmbeddedWebviewController *embeddedWebviewController
-       = [[MSIDAADOAuthEmbeddedWebviewController alloc] initWithStartURL:configuration.startURL
-                                                                  endURL:[NSURL URLWithString:configuration.endRedirectUrl]
-                                                                 webview:webview
-                                                           customHeaders:configuration.customHeaders
-                                                          platfromParams:platformParams
-                                                                 context:context];
+#if MSAL_JS_AUTOMATION
```

Comment:
Instead of all those #if - #elses, would it be perhaps cleaner to have a writable property in each of those classes (wrapped by ifdef) and assign it after instance creation?